### PR TITLE
adds feature that lets defining specific lifetime for a route

### PR DESCRIPTION
**Pull request by vjolenz**

### CODE REVIEW

**Comment by freekmurze** [comment]
Nice, could you also add a test the proves that it will not use the cache after the set lifetime?

You could use Carbon's `setTestNow` function for this.

**Comment by vjolenz** [comment]
@freekmurze I tried using Carbon's setTestNow function but it seems it doesnt have any affect on cache

Example snippet:
```
Cache::put('test', 'testval', 5);
Carbon::setTestNow(Carbon::now()->addMinutes(7));
dd(Cache::get('test'));
```

When I run this snippet, I still get testval.

**Comment by vjolenz** [comment]
https://github.com/spatie/laravel-responsecache/blob/ba18210c3c42d0eb5ddf805bf55e20eb435e6206/tests/TestCase.php#L181
This line makes trouble for this PR. If CacheResponse middleware is pushed to Kernel, it already caches the response. So, I think this line should be edited but I got no idea about how to solve for now

**Comment by vjolenz** [comment]
Solution was on the screen all the time but I couldn't realize :) I hope it is fine now.

File: tests/IntegrationTest.php [hunk 1]
```diff
@@ -173,4 +174,44 @@ public function it_will_cache_file_responses()
...
+    }
+
+    /** @test */
+    public function it_will_reproduce_cache_if_given_lifetime_is_expired()
+    {
+        // Set default lifetime as 0 to disable middleware that is already pushed to Kernel
+        $this->app['config']->set('responsecache.cache_lifetime_in_minutes', 0);
+
+        Carbon::setTestNow(Carbon::now()->subMinutes(6));
+        $firstResponse = $this->call('get', '/cache-for-given-lifetime');
```
Comment (freekmurze) [inline tests/IntegrationTest.php]: Can you call it for a second time here and assert that it now has cached the result?

Comment (vjolenz) [reply tests/IntegrationTest.php]: Done 👍

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**